modules/library/main/src/main/java/org/geotools/filter/function/NowFunction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ */
+package org.geotools.filter.function;
+
+import java.util.Date;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+/** Return the current time (milliseconds) as a Date */
+public class NowFunction extends FunctionExpressionImpl {
```

Comment:
Should this not implement `VolatileFunction` otherwise you run the risk of it getting cached

Reply:
It does not, otherwise different features would be compared to different times. From the javadoc of VolatileFunction:

```
Functions whose value changes over time but not within the same feature collection filtering
are considered to be <em>stable</em> and as such their result can be considered a constant during
the single data access operation
```

There is no cross-request caching, the SimplifyingFilterVisitor transforms filters incoming from the user request on the fly each time.

Reply:
OK, but I would make that clear in the documentation - I would expect to get a different value back each time I called a function called Now :-)

Reply:
It does give a different value... except when it's optimized out in SimplifyingFilterVisitor. It does that for every function not taking arguments, or taking static arguments. But yes, does not hurt to remind it in its javadoc.

Reply:
I have updated the javadoc, what do you think @ianturton ?